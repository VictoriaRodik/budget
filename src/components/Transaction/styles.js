import styled from "styled-components";

export const Wrapper = styled.div`
  background: ${({ value, theme }) =>
    value < 0 ? theme.additionalBackground : theme.background};
  color: ${({ theme }) => theme.color};
  border: 1px solid;
  border-color: ${({ theme }) => theme.color};
  border-radius: 3px;
  padding: 15px;
  margin-bottom: 10px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  font-size: 16px;

  @media (min-width: 480px) {
    padding: 10px;
    font-size: 18px;
  }

  @media (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }
`;

Wrapper.displayName = "TransactionWrapper";

export const TransactionDate = styled.div`
  flex-grow: 1;
`;
TransactionDate.displayName = "TransactionDate";

export const Value = styled.div`
  flex-grow: 1;
  margin: 0 0 5px 30px;
`;
Value.displayName = "Value";

export const Comment = styled.div`
  flex-grow: 2;
  margin: 0 0 5px 30px;
  transition: opacity 0.3s ease;
  opacity: ${({ isHovered }) => (isHovered ? 1 : 0.7)};
  &:hover {
    opacity: 1;
  }
`;
Comment.displayName = "Comment";

export const Icon = styled.span`
  img {
    max-width: 20px;
    margin-right: 10px;
  }
`;
Icon.displayName = "Icon";

export const DateContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-bottom: 5px;
`;
DateContainer.displayName = "DateContainer";
