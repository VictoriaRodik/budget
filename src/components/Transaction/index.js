import PropTypes from "prop-types";
import { Wrapper, TransactionDate, Value, Comment } from "./styles";
import { AppContext } from "../providers/context";
import { useCallback, useContext } from "react";

const Transaction = ({
  transaction: { id, value, date, comment },
  onDelete,
}) => {
  const { state } = useContext(AppContext);

  const deleteItem = useCallback(() => onDelete(id), [id]);

  return (
    <Wrapper value={value}>
      <TransactionDate>{date}</TransactionDate>
      <Value name="value">
        {value.toFixed(2)}, {state.currency}
      </Value>
      <Comment>{comment}</Comment>
      <button onClick={deleteItem}>Remove</button>
    </Wrapper>
  );
};

Transaction.propTypes = {
  transaction: PropTypes.shape({
    date: PropTypes.string,
    value: PropTypes.number,
  }),
};

Transaction.defaultProps = {
  transaction: {
    date: "",
    value: 0,
  },
};

export default Transaction;
