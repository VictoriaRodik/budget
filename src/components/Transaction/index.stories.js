import Transaction from '.'

export default {
  title: 'Example/Transaction',
  component: Transaction,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/configure/story-layout
    layout: 'fullscreen',
  },
};

export const Income = {
  args: {
    transaction: {
      date: '2024-01-23',
      value: 10000,
      comment: 'Salary'
    },
  },
};


