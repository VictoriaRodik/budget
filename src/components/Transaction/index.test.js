import React from "react";
import Transaction from ".";
import renderer from 'react-test-renderer';

describe('Transaction component', () => {
    let component;
    let props;

    beforeEach(() => {
        props = {
            transaction: {
                value: 100,
                comment: 'test',
                date: '01.01.2024'
            }
        };
        component = renderer.create(<Transaction {...props} />);
    })

    it('should show transaction', () => {
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});