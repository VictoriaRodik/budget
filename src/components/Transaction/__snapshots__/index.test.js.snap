// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should show transaction 1`] = `
<div
  className="sc-bdfCDU fjbXMR"
  value={100}
>
  <div
    className="sc-gsTDqH jCRJqK"
  >
    01.01.2024
  </div>
  <div
    className="sc-dlfmHC czogkz"
    name="value"
  >
    100.00
  </div>
  <div
    className="sc-hKgKIp eIcyOC"
  >
    test
  </div>
</div>
`;
